pkg/controller/nodenetworkconfigurationpolicy/selectors/manager.go

```diff
@@ -0,0 +1,54 @@
+package selectors
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+
+	nmstatev1alpha1 "github.com/nmstate/kubernetes-nmstate/pkg/apis/nmstate/v1alpha1"
+)
+
+type Manager struct {
+	client   client.Client
+	policy   nmstatev1alpha1.NodeNetworkConfigurationPolicy
+	nodeName string
+	logger   logr.Logger
+}
+
+func NewManager(client client.Client, nodeName string, policy nmstatev1alpha1.NodeNetworkConfigurationPolicy) Manager {
```

Comment:
Could you please elaborate why did you chose to have a stateful manager for this over a simple input-output function? That would be easier to test, all we need is `func Matches(policy, nodeName) (bool, string, error) {}`. Bool for whether it matches, string for the message `"All matches"`/`"Following does not match:\n..."` and error for unexpected failures. Maybe I'm missing something and there is a good reason to have a manager abstratction. Otherwise, I prefer simplicity.

Reply:
The approach I was taking is having a Matching method per type, for example:
- `UnmatchedLabels`
- `UnmatchedNodeAffinity`
- `UnmatchedNetworkAffinity`

They don't do string formating inside they just return what is not matching and the conditions.Manager is the one that know how to format that into a condition.

The controller use ther results from selectors.Manager and calls conditions.Manager with them.

So controller calls them and pass result to conditions.Manager functions to compose proper condition.

Reply:
It is good to look forward to the new features, but I have a question. What would those method share that would be a part of the context. Could they live under separated packages (to limit the scope, make modules shorter and easier to learn and grasp the logic).

I also don't get the relation to conditions.Manager you explained. This module and conditions should live without knowing about one another, they should be connected through the main module.

++ for the UnmatchedLabels function, I like it.

Reply:
> It is good to look forward to the new features, but I have a question. What would those method share that would be a part of the context. 

The three of them will need client/policy/nodeName and that's what Manager struct has more or less.

> Could they live under separated packages (to limit the scope, make modules shorter and easier to learn and grasp the logic).

If we move to modules we cannot create methods there for Manager since they are not local to module, we can just split manager.go at labels.go, node_affinity.go, network_affinity.go and leave at manager.go only the struct.

Reply:
Also I don't really like the work Manager for a struct, since at golang they don't own the methods, they are just structs, their name has to be associated with the content.

Reply:
> If we move to modules we cannot create methods there for Manager since they are not local to module

I know, I was more in favor of just dropping the manager and keep simple functions. However, if you believe the manager makes the code simpler to read, let's keep it. About the name, maybe express that you are working with selectors in it?

Reply:
@phoracek maybe this https://github.com/nmstate/kubernetes-nmstate/pull/285/commits/4bb973b6a93f69896ac82dc0861432c3b4e2e3c3 make more sense ?